sds/src/path.rs

```diff
@@ -50,6 +51,23 @@ impl<'a> Path<'a> {
         }
         true
     }
+
+    pub fn sanitize(&self) -> String {
```

Comment:
Do you think this is a good refactoring? [Improve sanitize algorithm](https://github.com/DataDog/dd-sensitive-data-scanner/pull/53/commits/18216516b958e2fd793230fb709bd380189cde16)

Reply:
The changes you made are all good. The `field.chars().collect()` is still going to generate quite a few unnecessary memory allocations though. I would refactor `standardize_path_chars` to accept a `&str` instead of `Vec<char>`, and that should improve things quite a bit there.

The `standardize_path_chars` implementation was fine when it was just used for generating the regex during scanner creation, but now it's used during scanning which is much more performance sensitive.

Reply:
Ok I see, I'm doing the change

Reply:
[Avoid using Vec[char] in path standardization](https://github.com/DataDog/dd-sensitive-data-scanner/pull/53/commits/53900ad8afc3cefd8f0b8bc1cba00333a0b1292d)

I'm merging to be able to test on staging today, but feel free to tell me if something can still be improved here 🙇